File: LeetCodeCPlusPlus/py11.cpp
#include<vector>
#include<iostream>
using namespace std;
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0, right = height.size()-1, ans;
        ans = (right-left)*min(height[right], height[left]);
        while (left<right) {
            if (height[left]<height[right]) 
                left++;
            else
                right--;
            ans = max(ans, (right-left)*min(height[right], height[left]));
        }
        return ans;
    }
};

File: LeetCodePython/py11.py
# 盛最多水的容器
class Solution:
    def maxArea(self, height: List[int]) -> int: 
        left = 0; right = len(height)-1
        ans = (right-left)*min(height[left], height[right])
        while  left<right:
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
            hold = (right-left)*min(height[left], height[right])
            ans = max(ans, hold)
            
        return ans